language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -171,6 +174,8 @@ public ExtendedLanguageClient getClient() {
             experimentalClientCapabilities = new Gson().fromJson(params.getCapabilities().getExperimental().toString(),
                     HashMap.class);
         }
+        // Subscribe for document sync events
+        subscribeDocumentSyncEvents();
```

Comment:
Here we are invoking individual events. When we are going to add more publishers we have to add more and more private APIs. Instead, we can use encapsulation and expose separate APIs via a single object. Since we have `SubscribersHolder` already we can use it as a more generic encapsulated API.